test/stubdata/producer_interceptor_stub.go

```diff
@@ -0,0 +1,21 @@
+package test
+
+import "github.com/Trendyol/kafka-konsumer/v2"
+
+type MockProducerInterceptor struct{}
+
+const (
+	XSourceAppKey   = "x-source-app"
+	XSourceAppValue = "kafka-konsumer"
+)
+
+func (i *MockProducerInterceptor) OnProduce(ctx kafka.ProducerInterceptorContext) {
```

Comment:
I think, stubdata directory is not necessary, we can put this implementation on integration_test